crates/mun_runtime/src/assembly.rs

```diff
@@ -40,68 +42,131 @@ impl Assembly {
         set_allocator_handle(allocator_ptr);
 
         let info = get_info();
-
-        for function in info.symbols.functions() {
-            runtime_dispatch_table.insert_fn(function.signature.name(), function.clone());
-        }
-
-        Ok(Assembly {
+        let assembly = Assembly {
             library_path: library_path.to_path_buf(),
-            library: Some(library),
+            library,
+            legacy_libs: Vec::new(),
             info,
             allocator: gc,
-        })
+        };
+
+        assembly.ensure_linkable(runtime_dispatch_table)?;
```

Comment:
Wouldnt it be nicer to do this outside of the new call? That way you can also load an Assembly even though it doesn't link. We should obviously only take it in use if its actuall linkable but for the sake of single responsibility I think this call should be outside of the `new` call.